protostar-rust/src/main.rs

```diff
@@ -66,16 +77,38 @@ fn dependencies_for_package(
 fn main_execution() -> Result<()> {
     let _args = Args::parse();
 
+    let project_root_path = project_root::get_project_root()?
+        .to_str()
+        .expect("error reading project root path")
+        .to_owned();
     // TODO #1997
-    set_var("CARGO_MANIFEST_DIR", "../../cairo/Cargo.toml");
+    set_var(
+        "CARGO_MANIFEST_DIR",
+        format!("{}/../cairo/Cargo.toml", project_root_path),
+    );
 
     let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
 
+    let mut protostar_test_config = ProtostarTestConfig {
```

Comment:
As i mentioned in another PR, this is not a config, it is more of a TestData. Config suggests user configuration

Reply:
with new changes we also have `corelib_path` there which makes it a config I think.

Reply:
> we also have corelib_path there which makes it a config I think.
We shouldn't have it there, Cairo hint processor shouldn't do anything related to compilation.

Reply:
it is removed with building in scarb
[adapt for building in scarb](https://github.com/software-mansion/protostar/pull/2059/commits/b104f483c4ffb234aba0256cd0a8d988e727484b)